Test utils: fix daemon start utils

coolljt0725 opened the pull request:
User call `Start` could with args, and the args could
contains `--storage-driver`, but in `Start`, it will add
`--storage-driver` even though user has specified `--storage-driver`
in args.
hit this issues when I working on PR #19711 

Signed-off-by: Lei Jitang leijitang@huawei.com

runcom commented:
SGTM

duglin commented on integration-cli/docker_utils.go:
```diff
@@ -225,16 +225,19 @@ func (d *Daemon) Start(arg ...string) error {
 	// If we don't explicitly set the log-level or debug flag(-D) then
 	// turn on debug mode
 	foundIt := false
```
@coolljt0725 can you do me favor and rename `foundIt` to something like `foundLog`? Since we have more than one "found" variable now I think we need to make "It" be more explicit.

duglin commented:
LGTM - just one minor ask - if not done here I'll deal with it in another PR.

coolljt0725 commented:
@duglin  done :smile:

vdemeester commented:
LGTM 🐲

coolljt0725 committed:
Test utils: fix daemon start utils

User call `Start` could with args, and the args could
contains `--storage-driver`, but in `Start`, it will add
`--storage-driver` even though user has specified `--storage-driver`
in args.

Signed-off-by: Lei Jitang <leijitang@huawei.com>

duglin commented:
@coolljt0725 thanks a ton!
LGTM once janky is done
